AvrDataTypes/main/lib/AvrDataTypes.c

```diff
@@ -88,17 +88,59 @@ char* substring(char* str, uint8_t start, uint8_t end) {
     return buffer;
 }
 
-volatile typeof(uint8_t) PORT[] = {0, 1, 2, 3, 4, 5, 6, 7};
-
 /**
- * @brief Gets the Length of array [PORT*] from its size.
- * 
- * @param PORT[] change the PORT array and re-run this to get the length of the new PORT array
- * @return size_t 
+ * @brief Converts a register to an array logic of bits using utility methods.
  */
-size_t getLength() {
-        return sizeof(PORT) / sizeof(typeof(*PORT));
-}
+namespace Register {
+
+    const static uint8_t PORT[] = {0, 1, 2, 3, 4, 5, 6, 7};
+    const static uint8_t HIGH = 0b01;
+    const static uint8_t LOW = 0b00;
+
+    struct Register8 {
+
+        /**
+         * @brief Gets the Length of array [PORT*] from its size.
+         * 
+         * @param PORT[] change the PORT array and re-run this to get the length of the new PORT array
+         * @return size_t 
+         */
+        size_t getLength() {
+            return sizeof(PORT) / sizeof(typeof(*PORT));
+        }
+
+        /**
+         * @brief Digitally writes a [STATE] to a [PIN] in a particular [PORT].
+         * 
+         * @param PORT the register
+         * @param PIN the pin
+         * @param STATE the STATE, either [Regiter::HIGH] or [Register::LOW]
+         */
+        void digitalWrite(volatile uint8_t& PORT, const uint8_t& PIN, const uint8_t& STATE) {
```

Comment:
Here is how the `Register::Register8::digitalWrite(...)` works: 
1) 3 inputs, `PORT` representing the register to toggle a pin in, `PIN` representing a pin (bit) number from 0 to 7 to toggle inside the register, a `STATE` either `Register::HIGH` or `Register::LOW`.
2) Find the `BIT` binary representation by shifting the HIGH (0b01) to the location of the pin (PIN).
3) If (digitalHigh); then -> PORT = (PORT) + (BIT), add the BIT to the PORT.
4) If (digitalLow); then -> PORT = (PORT) * !(BIT), flip the BIT and multiply it with the PORT, so that we multiply (0) only at the bit location of the PORT.